### DISCUSSION TITLE
SigStore KMIP integration

### DISCUSSION THREAD

**Discussion by daveroche-digi** [sigstore/sigstore#776] [Ideas]
Hi,
 
I would like to enquire if SigStore has any plans to facilitiate integrating technology partners to act as KMS service providers via the the KMIP standard. 

I note Sigstore has integrations in custom way with KMS providers for key storage using proprietary vender specific integrations which are tightly coupled to Sigstore e.g. Azure Key Vault, Hashicorp Vault, AWS KMS, and GCP KMS. We would like to enquire and discover if Sigstore intend to support a more general integration for KMS providers by leveraging an open standard like KMIP to facilitate other providers instead of integrating new partners individually in bespoke way. 

We feel it may be mutually beneficial as this opens up the possibility for more KMS services to connect with Sigstore service(s) using KMIP for services that store keys in a KMS service provider and are KMIP ready. 

Thanks, 

Dave

**Comment by bobcallaway** [comment]
I'd be supportive of adding a new provider for this, though at first I'd target it to be additive instead of replacing the existing implementations.

**Reply by daveroche-digi** [reply]
Thank you Bob, that makes total sense. And we would be happy to collaborate and beta test this in support.

**Reply by znewman01** [reply]
I'm +1 here, too. Filed an issue to track: https://github.com/sigstore/sigstore/issues/784
